src/main/java/eu/webeid/security/validator/certvalidators/SubjectCertificatePurposeValidator.java

```diff
@@ -46,9 +46,18 @@ public final class SubjectCertificatePurposeValidator {
      */
     public static void validateCertificatePurpose(X509Certificate subjectCertificate) throws AuthTokenException {
         try {
+            final boolean[] keyUsage = subjectCertificate.getKeyUsage();
+            if (keyUsage == null) {
+                throw new UserCertificateMissingPurposeException();
+            }
+            if (!keyUsage[KEY_USAGE_DIGITAL_SIGNATURE]) {
+                throw new UserCertificateWrongPurposeException();
+            }
             final List<String> usages = subjectCertificate.getExtendedKeyUsage();
             if (usages == null || usages.isEmpty()) {
-                throw new UserCertificateMissingPurposeException();
+                // Digital Signature extension present, but Extended Key Usage extension not present,
+                // assume it is an authentication certificate (e.g. Luxembourg eID).
+                return;
```

Comment:
Line 68 has a debug log message, it is needed here as well before return for consistency.